addons/mail/static/tests/owl/component/follower_tests.js

```diff
@@ -0,0 +1,383 @@
+                return { follower };
+            });
+        }
+    }
+    ParentComponent.env = this.env;
+    Object.assign(ParentComponent, {
+        components: { Follower },
+        props: { followerLocalId: String },
+        template: xml`<div>
+            <p>parent</p>
+            <t t-if="storeProps.follower">
+                <Follower followerLocalId="props.followerLocalId"/>
+            </t>
+        </div>`,
+    });
+    const component = new ParentComponent(null, { followerLocalId });
+    await component.mount(this.widget.el);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Follower').length,
+        1,
+        "should have follower component"
+    );
+    assert.strictEqual(
+        document.querySelectorAll('.o_Follower_removeButton').length,
+        1,
+        "should display a remove button"
+    );
+
+    await document.querySelector('.o_Follower_removeButton').click();
+    assert.verifySteps(
+        ['message_unsubscribe'],
+        "clicking on remove button should call 'message_unsubscribe' route");
+    await afterNextRender();
+    assert.strictEqual(
+        document.querySelectorAll('.o_Follower').length,
+        0,
+        "should no more have follower component"
+    );
+    component.destroy();
+});
+
+QUnit.test('click on edit follower', async function (assert) {
+    assert.expect(6);
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === 'web/image/res.partner/3/image_128') {
+                return;
+            }
+            if (route.includes('/mail/read_subscription_data')) {
+                assert.step('fetch_subtypes');
+                return [{
+                    default: true,
+                    followed: true,
+                    internal: false,
+                    id: 1,
+                    name: "Dummy test",
+                    res_model: 'res.partner'
+                }];
+            }
+            return this._super(...arguments);
+        },
+    });
+    const followerLocalId = await this.env.store.dispatch('_createFollower', {
+        partner_id: this.env.session.partner_id,
+        email: "bla@bla.bla",
+        id: 2,
+        is_active: true,
+        is_editable: true,
+        name: "François Perusse",
+    }, 'dummy_and_useless_thread_local_id');
+    await this.createFollowerComponent(followerLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Follower').length,
+        1,
+        "should have follower component"
+    );
+    assert.strictEqual(
+        document.querySelectorAll('.o_Follower_editButton').length,
+        1,
+        "should display an edit button"
+    );
+
+    await document.querySelector('.o_Follower_editButton').click();
+    await afterNextRender();
+    assert.verifySteps(['fetch_subtypes'], 'clicking on edit follower should fetch subtypes');
+
+    // Check the dialog has been created (dialogService does not work in tests)
```

Comment:
should make it work. that's because it is not initialized properly by simulating that webclient is ready.
chat window service works thanks to this: https://github.com/odoo-dev/odoo/blob/master-mail-owl-xdu/addons/mail/static/tests/owl/test_utils.js#L126